src/components/chat/GithubRepoModal.tsx

```diff
@@ -0,0 +1,346 @@
+    }
+  };
+
+  const handleSelectRepo = async (repo: any) => {
+    setImporting(true);
+    setError(null);
+    try {
+      const result = await IpcClient.getInstance().cloneRepoFromUrl({
+        url: `https://github.com/${repo.full_name}.git`,
+        installCommand: installCommand.trim() || undefined,
+        startCommand: startCommand.trim() || undefined,
+      });
+
+      if (result.success && result.app) {
+        setSelectedAppId(result.app.id);
+        onClose();
+
+        await checkAndGenerateAiRules(result.app.id, result.app.path);
+      } else {
+        setError(result.error || "Failed to import repository.");
+      }
+    } catch (err: any) {
+      setError(err.message || "Failed to import repository.");
+    } finally {
+      setImporting(false);
+    }
+  };
+
+  const hasInstallCommand = installCommand.trim().length > 0;
+  const hasStartCommand = startCommand.trim().length > 0;
+  const commandsValid = hasInstallCommand === hasStartCommand;
+
+  // Show GitHub connection UI if not authenticated
+  if (!isAuthenticated) {
+    return (
+      <Dialog open={isOpen} onOpenChange={onClose}>
+        <DialogContent className="max-w-2xl">
+          <DialogHeader>
+            <DialogTitle className="flex items-center gap-2">
+              <Github className="w-5 h-5" /> Connect to GitHub
+            </DialogTitle>
+            <DialogDescription>
+              Connect your GitHub account to import repositories into Dyad.
+            </DialogDescription>
+          </DialogHeader>
+          <UnconnectedGitHubConnector
+            appId={null}
+            folderName=""
+            settings={settings}
+            refreshSettings={refreshSettings}
+            handleRepoSetupComplete={() => undefined}
+            expanded={false}
+          />
+        </DialogContent>
+      </Dialog>
+    );
+  }
+
+  return (
+    <Dialog open={isOpen} onOpenChange={onClose}>
+      <DialogContent className="max-w-2xl">
+        <DialogHeader>
+          <DialogTitle className="flex items-center gap-2">
+            <Github className="w-5 h-5" /> Import from GitHub
+          </DialogTitle>
+          <DialogDescription>
+            Clone a repository from GitHub to start working on it in Dyad.
+          </DialogDescription>
+        </DialogHeader>
+
+        <Alert className="border-blue-500/20 text-blue-500">
+          <Info className="h-4 w-4" />
+          <AlertDescription>
+            After cloning, Dyad will automatically generate an AI_RULES.md file
+            if one doesn't exist.
+          </AlertDescription>
+        </Alert>
+
+        <Tabs defaultValue="your-repos" className="w-full">
+          <TabsList className="grid w-full grid-cols-2">
+            <TabsTrigger value="your-repos">Your Repositories</TabsTrigger>
+            <TabsTrigger value="from-url">From URL</TabsTrigger>
+          </TabsList>
+
+          <TabsContent value="your-repos" className="space-y-4">
+            {loading && (
```

Comment:
if you've already loaded the repo before (e.g. opened dialog, close dialog and re-open dialog), it's a bit jarring to have a loading spinner that then closes. Can we instead put the spinner next to the tab header? What do you think?